packages/common/driver-definitions/src/factory.ts

```diff
@@ -0,0 +1,24 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Factory options interface which are going to be used in the loader layer.
+ */
+export interface IDriverFactoryLoaderOptions {
+	[DriverFactoryLoaderOptions.summarizeProtocolTree]: boolean;
+}
+
+export interface IDriverFactoryOptions extends Partial<IDriverFactoryLoaderOptions> {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	[index: string]: any;
```

Comment:
I'm not sure there is a value here to more clearly describe extensibility mechanism here. We can leave with fixed IDriverFactoryLoaderOptions  and keep adding new options over time just fine.